src/Controller/ProfileController.php

```diff
@@ -124,6 +125,17 @@ public function editAction(Request $request): Response
         $form->handleRequest($request);
 
         if ($form->isSubmitted() && $form->isValid()) {
+            $uow = $this->getEM()->getUnitOfWork();
+            $uow->computeChangeSets();
+            $changeSet = $uow->getEntityChangeSet($user);
+
+            $diffs = array_intersect(array_keys($changeSet), ['email', 'username']);
+
+            if (!empty($diffs) && $user instanceof User) {
+                $reason = sprintf('Your %s has been changed', implode(' and ', $diffs));
+                $userNotifier->notifyChange($changeSet['email'][0] ?? $user->getEmail(), $reason);
```

Comment:
I think it would be nice to send the email to both old and new emails here when the email changes.